addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2659,14 +2659,36 @@ export class OdooEditor extends EventTarget {
         };
 
         const side1 = isRow ? 'left' : 'top';
+        const side2 = isRow ? 'top' : 'left';
+        const scrollContainerRect = this.options.getScrollContainerRect();
+
         ui.style[side1] = (isRow ? elementRect : tableRect)[props.xy[side1]] - togglerRect[props.size[side1]] + 'px';
-        ui.style[props.size[side1]] = !isRow && togglerRect[props.size[side1]] + 'px';
+        if (tableRect.top > scrollContainerRect.top) {
+            ui.style[props.size[side1]] = !isRow && togglerRect[props.size[side1]] + 'px';
+            if (tableRect[side1] - scrollContainerRect.top < 15 && !isRow) {
+                ui.style[props.size[side1]] = wrappedUi.style[props.size[side1]] = tableRect[side1] - scrollContainerRect.top + 'px';
+                ui.style[side1] = tableRect[side1] - parseFloat(wrappedUi.style[props.size[side1]]) + 'px';
+            } else {
+                ui.style[props.size[side1]] = wrappedUi.style[props.size[side1]] = '15px';
+            }
 
-        const side2 = isRow ? 'top' : 'left';
-        wrappedUi.style[props.size[side2]] = elementRect[props.size[side2]] + 'px';
-        ui.style[side2] = tableRect[props.xy[side2]] + 'px';
-        wrappedUi.style[side2] = elementRect[side2] - tableRect[side2] - 1 + 'px';
-        ui.style[props.size[side2]] = tableRect[props.size[side2]] + 'px';
+            wrappedUi.style[props.size[side2]] = elementRect[props.size[side2]] + 'px';
+            ui.style[side2] = tableRect[props.xy[side2]] + 'px';
+            wrappedUi.style[side2] = elementRect[side2] - tableRect[side2] - 1 + 'px';
+            ui.style[props.size[side2]] = tableRect[props.size[side2]] + 'px';
+        } else if (tableRect.top < scrollContainerRect.top + togglerRect['height']) {
```

Comment:
How about a doc string something like : 'When table is not completely in viewport'